app/scripts/controllers/groups/TransferClientsController.js

```diff
@@ -5,11 +5,27 @@
             scope.tempData = [];
             scope.selectedClients = [];
             scope.selectedMembers = [];
-            resourceFactory.groupResource.get({paged: 'true', orderBy: 'name', sortOrder: 'ASC'}, function (data) {
-                scope.groups = _.reject(data.pageItems, function (group) {
-                    return group.id == routeParams.id;
+            scope.formData = {};
+            scope.destinationGroup = "";
+
+            scope.groups = function(value){
+                var jsonList = [];
+                resourceFactory.groupResource.getAllGroups({sqlSearch: value ,orderBy: 'name', sortOrder: 'ASC'}, function (data) {
+                    scope.data = data;
+                    scope.group = _.reject(scope.data, function (group) {
+                        return group.id == routeParams.id;
+                    });
                 });
-            });
+                for(var i in scope.group){
```

Comment:
why do we need to copy scope.groups to another array? why not simply return scope.groups ?